crates/p2p/src/tests/three_handlers.rs

```diff
@@ -0,0 +1,53 @@
+//! Three-handlers integration test for new handler-based architecture.
+
+use chrono as _;
+use tracing::info;
+
+use super::common::Setup;
+use crate::{commands::Command, events::ReqRespEvent};
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_reqresp_basic() -> anyhow::Result<()> {
+    let Setup {
+        cancel,
+        mut user_handles,
+        tasks,
+    } = Setup::all_to_all(2).await?;
```

Comment:
Let's try to add the `disconnect_from_all` method to `P2P` and call it at the end of the test to address the test panic.